### PR TITLE
[Docker-outside-of-docker] Increase backlog depth for socat proxy (#789)

### PR DESCRIPTION
**Pull request by groner**
The underlying problem described in #789 is that the main `accept(2)` loop in `socat` is single threaded and is easily overwhelmed by docker compose, which can create connection attempts in multiple cores.

Fortunately the listening socket already has a backlog which we can increase from [socat's default of 5](http://www.dest-unreach.org/socat/doc/socat.html#GROUP_LISTEN).

In this PR I have chosen a depth of 128 as probably high enough for most devcontainer needs. If it isn't high enough this could be made tunable, but reevaluating the proxy implementation might be in order also.

### CODE REVIEW

**Comment by eljog** [comment]
Thank you for the contribution.
I can approve to run the pipeline checks once the CLA has been accepted

**Comment by groner** [comment]
@microsoft-github-policy-service agree

**Comment by groner** [comment]
@eljog Thanks for looking at this.

I've added some details to the description, but refer to #789 for context. I hope that's alright.

**Comment by eljog** [comment]
I have approved to run the workflow checks

File: src/docker-outside-of-docker/install.sh [hunk 1]
```diff
@@ -329,7 +329,7 @@ if [ "${ENABLE_NONROOT_DOCKER}" = "true" ] && [ "${SOURCE_SOCKET}" != "${TARGET_
             log "Enabling socket proxy."
             log "Proxying ${SOURCE_SOCKET} to ${TARGET_SOCKET} for vscode"
             sudoIf rm -rf ${TARGET_SOCKET}
-            (sudoIf socat UNIX-LISTEN:${TARGET_SOCKET},fork,mode=660,user=${USERNAME} UNIX-CONNECT:${SOURCE_SOCKET} 2>&1 | sudoIf tee -a \${SOCAT_LOG} > /dev/null & echo "\$!" | sudoIf tee \${SOCAT_PID} > /dev/null)
+            (sudoIf socat UNIX-LISTEN:${TARGET_SOCKET},fork,mode=660,user=${USERNAME},backlog=128 UNIX-CONNECT:${SOURCE_SOCKET} 2>&1 | sudoIf tee -a \${SOCAT_LOG} > /dev/null & echo "\$!" | sudoIf tee \${SOCAT_PID} > /dev/null)
```
Comment (eljog) [inline src/docker-outside-of-docker/install.sh:332]: LGTM
@joshspicer any concerns with raising the backlog to 128 (from default 5)?

**Review by eljog — APPROVED** [review]

File: src/docker-outside-of-docker/install.sh (hunk 1 above)
Comment (joshspicer) [reply src/docker-outside-of-docker/install.sh:332]: No concern here, looks reasonable and passes the current checks!

**Pull request merged**